tests/time_tests/test_runner/conftest.py

```diff
@@ -133,22 +133,43 @@ def temp_dir(pytestconfig):
 
 
 @pytest.fixture(scope="function")
-def cl_cache_dir(pytestconfig):
+def cl_cache_dir(pytestconfig, instance):
     """Generate directory to save OpenCL cache before test run and clean up after run.
 
     Folder `cl_cache` should be created in a directory where tests were run. In this case
     cache will be saved correctly. This behaviour is OS independent.
     More: https://github.com/intel/compute-runtime/blob/master/opencl/doc/FAQ.md#how-can-cl_cache-be-enabled
     """
-    cl_cache_dir = pytestconfig.invocation_dir / "cl_cache"
-    # if cl_cache generation to a local `cl_cache` folder doesn't work, specify
-    # `cl_cache_dir` environment variable in an attempt to fix it (Linux specific)
-    os.environ["cl_cache_dir"] = str(cl_cache_dir)
-    if cl_cache_dir.exists():
+    if instance["device"]["name"] == "GPU":
+        cl_cache_dir = pytestconfig.invocation_dir / "cl_cache"
+        # if cl_cache generation to a local `cl_cache` folder doesn't work, specify
+        # `cl_cache_dir` environment variable in an attempt to fix it (Linux specific)
+        os.environ["cl_cache_dir"] = str(cl_cache_dir)
+        if cl_cache_dir.exists():
+            shutil.rmtree(cl_cache_dir)
+        cl_cache_dir.mkdir()
+        logging.info("cl_cache will be created in {}".format(cl_cache_dir))
+        yield cl_cache_dir
         shutil.rmtree(cl_cache_dir)
-    cl_cache_dir.mkdir()
-    yield cl_cache_dir
-    shutil.rmtree(cl_cache_dir)
+    else:
+        yield None
+
+
+@pytest.fixture(scope="function")
+def model_cache_dir(pytestconfig, executable):
+    """
+    Generate directory to IE model cache before test run and clean up after run.
+    """
+    if executable.stem == "timetest_infer_cache":
```

Comment:
Test runner shoudl be test agnostic. Can we pass cache type as a configuration parameter or option? This is a blocking comment

Reply:
Decided to create models_cache directory for any executable, and the executable will or won't use folder for cache creation. In case when "cached" runner replaces "default", this will make it possible not to change the logic

Reply:
My previous decision can't be achieved because we also need to check cache is created, so it can't be a common implementation for each case. So I added  `use_model_cache` field in new test_config.yml for `timetest_infer_cache`